etl/helpers.py

```diff
@@ -205,44 +237,33 @@ def create_dataset(
     # create new dataset with new metadata
     ds = catalog.Dataset.create_empty(dest_dir, metadata=default_metadata)
 
+    ds = _set_metadata_from_dest_dir(ds, dest_dir)
+
     # add tables to dataset
     used_short_names = set()
     for table in tables:
         if underscore_table:
             table = catalog.utils.underscore_table(table, camel_to_snake=camel_to_snake)
         if table.metadata.short_name in used_short_names:
             raise ValueError(f"Table short name `{table.metadata.short_name}` is already in use.")
-        used_short_names.add(table.metadata.short_name)
-        ds.add(table, formats=formats, repack=repack)
 
-    # set metadata from dest_dir
-    pattern = (
-        r"\/"
-        + r"\/".join(
-            [
-                f"(?P<channel>{'|'.join(CHANNEL.__args__)})",
-                "(?P<namespace>.*?)",
-                r"(?P<version>\d{4}\-\d{2}\-\d{2}|\d{4}|latest)",
-                "(?P<short_name>.*?)",
-            ]
-        )
-        + "$"
-    )
-
-    match = re.search(pattern, str(dest_dir))
-    assert match, f"Could not parse path {str(dest_dir)}"
+        if ds.metadata.channel == "grapher":
+            # if a table contains `date` instead of `year`, adapt it for grapher
+            if "date" in table.all_columns and "year" not in table.all_columns:
+                if "date" in table.index.names:
+                    table = table.reset_index("date")
+                    table = gh.adapt_table_with_dates_to_grapher(table)
+                    table = table.set_index("year", append=True)
+                else:
+                    table = gh.adapt_table_with_dates_to_grapher(table)
```

Comment:
Very nice to have out-of-the-box support! Thanks

In COVID, I've been manually adapting the columns to have `year` instead of `date`, where `year` would be 'days since zero_day`:

https://github.com/owid/etl/blob/5c62946e9451e3f62b3b308aee31644e4af25759/etl/steps/data/grapher/covid/latest/cases_deaths.py#L24

using function from

https://github.com/owid/etl/blob/5c62946e9451e3f62b3b308aee31644e4af25759/etl/steps/data/grapher/covid/latest/shared.py#L33-L47

I suppose this is no longer needed? Do you know if your changes could break this?

Reply:
I also didn't know we already have a function `adapt_table_with_dates_to_grapher` for this. It might make sense to just get rid of all `yearIsDay` in our codebase and let it all happen automatically.

> Do you know if your changes could break this?

I hope not. If it causes problems, let me know and I'll look into it.